KitePrintSDK/src/main/java/ly/kite/checkout/PaymentActivity.java

```diff
@@ -211,15 +262,45 @@ private void payWithNewCard() {
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_CODE_PAYPAL) {
             if (resultCode == Activity.RESULT_OK) {
-                PaymentConfirmation confirm = data.getParcelableExtra(com.paypal.android.sdk.payments.PaymentActivity.EXTRA_RESULT_CONFIRMATION);
-                if (confirm != null) {
+
+                PaymentConfirmation paymentConfirmation = data.getParcelableExtra( com.paypal.android.sdk.payments.PaymentActivity.EXTRA_RESULT_CONFIRMATION );
+
+                if (paymentConfirmation != null) {
+
+                    Analytics.getInstance( this ).trackPaymentCompleted( mPrintOrder, Analytics.PAYMENT_METHOD_PAYPAL );
+
                     try {
-                        String proofOfPayment = confirm.toJSONObject().getJSONObject("proof_of_payment").getJSONObject("adaptive_payment").getString("pay_key");
-                        submitOrderForPrinting(proofOfPayment);
-                    } catch (JSONException e) {
-                        showErrorDialog(e.getMessage());
+
+                      ProofOfPayment proofOfPayment = paymentConfirmation.getProofOfPayment();
+
+                      if ( proofOfPayment != null )
+                          {
+                          String paymentId = proofOfPayment.getPaymentId();
+
+                          //String proofOfPayment = paymentConfirmation.toJSONObject().getJSONObject("proof_of_payment").getJSONObject("adaptive_payment").getString( "pay_key" );
```

Comment:
I'm going to stop noting these unless you want me to continue -- You can deal with them at your leisure and definitely not a dependency to merging this request.